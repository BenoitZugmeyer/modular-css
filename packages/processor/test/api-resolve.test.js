"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".resolve()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should run resolvers until a match is found", () => {
            let ran = false;

            processor = new Processor({
                resolvers : [
                    () => {
                        ran = true;
                    },
                ],
            });

            expect(
                relative([
                    processor.resolve(
                        require.resolve("./specimens/start.css"),
                        "./local.css"
                    ),
                ])
            )
            .toMatchSnapshot();

            expect(ran).toBeTruthy();
        });

        it("should fall back to a default resolver", () => {
            processor = new Processor({
                resolvers : [
                    () => undefined,
                ],
            });

            expect(
                relative([
                    processor.resolve(
                        require.resolve("./specimens/start.css"),
                        "./local.css"
                    ),
                ])
            )
            .toMatchSnapshot();
        });
    });
});
