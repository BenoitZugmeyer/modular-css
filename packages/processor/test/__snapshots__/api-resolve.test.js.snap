// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .resolve() should fall back to a default resolver 1`] = `
Array [
  "packages/processor/test/specimens/local.css",
]
`;

exports[`/processor.js .resolve() should run resolvers until a match is found 1`] = `
Array [
  "packages/processor/test/specimens/local.css",
]
`;
